test/services/rules_cleanup_service_test.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class RulesCleanupServiceTest < ActiveSupport::TestCase
+  setup do
+    cp = FactoryBot.create(:canonical_profile, :with_rules)
+    acc = FactoryBot.create(:account)
+    FactoryBot.create(:profile, :with_rules, account: acc)
+    @unused_rules = cp.rules
+  end
+
+  test 'delete unused rules' do
```

Comment:
Would it be possible to cover more cases with tests please? ... to ensure that it does not delete stuff that it should not. :slightly_smiling_face:

Reply:
Do you mean adding `RuleReference` and `RuleIdentifier` records? Or something around rules/profiles that the current one is missing?

Reply:
Yes references, identifiers as well as profile rules, canonical profiles, non-canonical profiles, used vs. unused rules. Each `delete_all` has at least two scenarios (one which it deletes and one it should not delete)...

I see that we've added more of these scenarios to the test. Checking counts only might be a limiting factor as it does not fully check instances which are being/should be kept.

Reply:
Right, I'll add checks for that